24.md

```diff
@@ -17,7 +17,32 @@ These are extra fields not specified in NIP-01 that may be present in the string
   - `website`: a web URL related in any way to the event author.
   - `banner`: an URL to a wide (~1024x768) picture to be optionally displayed in the background of a profile screen.
   - `bot`: a boolean to clarify that the content is entirely or partially the result of automation, such as with chatbots or newsfeeds.
-  - `pronouns`: a string representing the preferred way to refer to this person.
+
+### User defined custom fields
+
+A `fields` attribute may also be added to profile metadata to represent user-defined fields. It is an array name-value tuples, eg:
+
+```json
+[["<name>", "<value>"], ...]
+```
+
+In the context of a full metadata object:
+
+```json
+{
+  "name": "Alex",
+  "picture": "https://...",
+  "fields": [
+    ["Pronouns", "ye/haw"],
+    ["Lifestyle", "vegan"],
+    ["Color", "green"]
+  ]
+}
+```
+
+While protocol-level fields like `lud16` and `picture` can influence the behavior of a user's account, custom fields are used for display purposes only. Clients should display custom fields in a confined area of the user's profile, and not make assumptions about their meaning.
```

Comment:
Why shouldn't custom feelds be used for more than display. What if i want to filter my feed for only displaying content from people who've provided prefered pronouns?

Reply:
If that is a use case, then `pronouns` should be back in the main object.

Reply:
With free-form fields, one user may have "Pronouns" another "Preferred pronouns" and so on. Even if you add functionality to filter on them it's not very thorough.

Reply:
> Why shouldn't custom feelds be used for more than display. What if i want to filter my feed for only displaying content from people who've provided prefered pronouns?

If this is functionality a client wants to implement they should just go ahead and do it using the main object, and if interoperability issues arise then we should define it in the spec.

One example is Monero people are already doing this to replace lightning addresses with Monero addresses in their own clients, yet don't require any definition in the main spec. Before we have hundreds of fields in the spec there should at least be some demand/usage in various clients

Imo